packages/ember-metal/tests/tracked/get_test.ts

```diff
@@ -0,0 +1,140 @@
+import { EMBER_METAL_TRACKED_PROPERTIES } from '@ember/canary-features';
+import { AbstractTestCase, moduleFor } from 'internal-test-helpers';
+import { get, getWithDefault, tracked } from '../..';
+
+if (EMBER_METAL_TRACKED_PROPERTIES) {
+  function createObj() {
+    class Obj {
+      @tracked string = 'string';
+      @tracked number = 23;
+      @tracked boolTrue = true;
+      @tracked boolFalse = false;
+      @tracked nullValue = null;
+      constructor() {
+        this.string = 'string';
+        this.number = 23;
+        this.boolTrue = true;
+        this.boolFalse = false;
+        this.nullValue = null;
+      }
+    }
+
+    return new Obj();
+  }
+
+  moduleFor(
+    '@tracked decorator: get',
+    class extends AbstractTestCase {
+      '@test should get arbitrary properties on an object'() {
+        let obj = createObj();
+
+        for (let key in obj) {
+          this.assert.equal(get(obj, key), obj[key], key);
+        }
+      }
+
+      '@test should retrieve a number key on an object'() {
+        class Obj {
+          @tracked 1 = 'first';
+          constructor() {
+            this[1] = 'first';
+          }
+        }
+
+        let obj = new Obj();
+
+        this.assert.equal(get(obj, '1'), 'first');
+      }
+
+      '@test should retrieve an empty key on an object'() {
+        class Obj {
+          @tracked '' = 'empty';
+          constructor() {
+            this[''] = 'empty';
+          }
+        }
+
+        let obj = new Obj();
+
+        this.assert.equal(get(obj, ''), 'empty');
+      }
+
+      '@test should get a @tracked path'() {
+        class Key {
+          @tracked value = 'value';
+          constructor() {
+            this.value = 'value';
```

Comment:
Why do we need to set this in the constructor if we're already initializing the property above?

Reply:
Fixed